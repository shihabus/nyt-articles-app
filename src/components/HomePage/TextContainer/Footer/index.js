/* eslint-disable camelcase */
import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import Icon from "react-native-vector-icons/FontAwesome";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between"
  },
  greyText: {
    fontSize: 14,
    color: "gray"
  },
  dateText: {
    fontSize: 12,
    color: "gray"
  },
  icon:{ alignSelf: "center" }
});

const Footer = ({ source, published_date }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.greyText} testID='Source Text'>{source}</Text>
      <Text style={styles.dateText} testID='Date container'>
        <Icon
          name="calendar"
          size={10}
          color="gray"
          style={styles.icon}
        />{" "}
        {published_date}
      </Text>
    </View>
  );
};

Footer.propTypes = {
  source: PropTypes.string,
  published_date: PropTypes.string
};

export default Footer;
