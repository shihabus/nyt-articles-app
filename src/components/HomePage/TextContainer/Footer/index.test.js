/* eslint-disable no-undef */
// /* eslint-disable import/named */
// /* eslint-disable no-undef */
// /**
//  * @jest-environment jsdom
//  */

import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Footer from "./index";
import { checkProps,findComponent } from "../../../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (props = {}) => {
  const wrapper = shallow(<Footer {...props} />);
  return wrapper;
};

describe("Footer", () => {
  let wrapper;
  beforeEach(() => {
    const props = {
        source: 'Test',
        published_date: 'Test'
    };
    wrapper = setUp(props);
  });

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot("Footer");
  });

  it("Checked props", () => {
    const expectedProps = {
        source: 'Test',
        published_date: 'Test'
    };
    const propsErr = checkProps(Footer, expectedProps);
    expect(propsErr).toBeUndefined();
  });

  it('should render child',()=>{
    expect(findComponent(wrapper,"Source Text").length).toEqual(1)
    expect(findComponent(wrapper,"Date container").length).toEqual(1)
  })
});
