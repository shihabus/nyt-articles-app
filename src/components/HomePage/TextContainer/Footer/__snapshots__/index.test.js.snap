// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot: Footer 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 14,
      }
    }
    testID="Source Text"
  >
    Test
  </Text>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 12,
      }
    }
    testID="Date container"
  >
    <Icon
      allowFontScaling={false}
      color="gray"
      name="calendar"
      size={10}
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
     
    Test
  </Text>
</View>
`;
